src/api/service/axios-bitbucket-client.ts

```diff
@@ -0,0 +1,73 @@
+import {AxiosInstance} from 'axios'
+import {ConsoleTransportOptions} from 'winston/lib/winston/transports'
+
+export const FIELDS = Object.freeze({
+  ISSUE_TYPE: 'issuetype',
+})
+export interface RepositoryReference {
+  slug: string;
+  project: {key: string};
+}
+export interface BranchReference {
+  name: string;
+  repository: RepositoryReference;
+}
+
+export interface CreatePullRequestOptions {
+  title: string;
+  fromRef: {id: string; repository: RepositoryReference};
+  toRef: {id: string; repository: RepositoryReference};
+}
+
+export class AxiosBitBucketClient {
+  readonly client: AxiosInstance
+
+  static parseUrl(url: string): RepositoryReference {
+    const regex = /https:\/\/(apps|bwa)\.nrs\.gov\.bc\.ca\/int\/stash\/projects\/(?<project>[^/]+)\/repos\/(?<repository>[^/\s]+)/gm
+    const m = regex.exec(url)
+    if (m === null) {
+      throw new Error(`Unable to parse BitBucket Url from ${url}`)
+    }
+    return {slug: m.groups?.repository as string, project: {key: m.groups?.project as string}}
+  }
+
+  constructor(client: AxiosInstance) {
```

Comment:
constructor(readonly client: AxiosInstance) {}

Declaration above is not needed.

Reply:
@mbystedt,  I am not sure what you are asking, if I remove that constructor I get a type script error:
```
src/api/service/axios-bitbucket-client.ts:22:12 - error TS2564: Property 'client' has no initializer and is not definitely assigned in the constructor.

22   readonly client: AxiosInstance
              ~~~~~~
```
Can you please clarify your suggestion?